lib/mayaUsd/ufe/private/Utils.cpp

```diff
@@ -129,54 +129,49 @@ UsdGeomXformCommonAPI convertToCompatibleCommonAPI(const UsdPrim& prim)
-                                         layerDisplayNames.c_str());
-        throw std::runtime_error(err.c_str());
-    }
-
-    // if the current layer doesn't have any contributions
-    if (!MayaUsdUtils::doesEditTargetLayerContribute(prim)) {
-        auto strongestContributingLayer = MayaUsdUtils::strongestContributingLayer(prim);
-        std::string err = TfStringPrintf("Cannot %s [%s]. It is defined on another layer. Please set [%s] as the target layer to proceed.", 
-                                         commandName.c_str(),
-                                         prim.GetName().GetString().c_str(),
-                                         strongestContributingLayer->GetDisplayName().c_str());
-        throw std::runtime_error(err.c_str());
-    }
-    else
+    // iterate over the prim stack, starting at the highest-priority layer.
+    for (const auto& spec : primStack) 
     {
-        auto layers = MayaUsdUtils::layersWithContribution(prim);
-        // if we have more than 2 layers that contributes to the final composed prim
-        if (layers.size() > 1) {
-            std::string layerDisplayNames;
+        const auto& layerName = spec->GetLayer()->GetDisplayName();
 
-            // skip the the first arc which is PcpArcTypeRoot
-            // we are interested in all the arcs after root
-            std::for_each(std::next(layers.begin()),layers.end(), [&](const auto& it) {
-                 layerDisplayNames.append("[" + it->GetDisplayName() + "]" + ",");
-            });
+        if (primSpec != spec) {
+            layerDisplayName.append("[" + layerName + "]" + ",");
+        }
+        else {
+            // if exist a primSpec with reference
+            if(spec->HasReferences()) {
+                break;
+            }
 
-            layerDisplayNames.pop_back();
-            std::string err = TfStringPrintf("Cannot %s [%s]. It has definitions or opinions on other layers. Opinions exist in %s",
-                                             commandName.c_str(),
-                                             prim.GetName().GetString().c_str(), 
-                                             layerDisplayNames.c_str());
-            throw std::runtime_error(err.c_str());
+            // if exists a def sepc
+            if (spec->GetSpecifier() == SdfSpecifierDef) {   
+                // if exists a def spec is in another layer other than current stage's local layer.
+                if(primSpec->GetLayer() != spec->GetLayer()) {
+                    layerDisplayName.append("[" + layerName + "]" + ",");
+                    break;
+                }
+                continue;
+            }
+
+            // if exists an over sepc
+            if (spec->GetSpecifier() == SdfSpecifierOver) {
+                layerDisplayName.append("[" + layerName + "]" + ",");
+            }
         }
     }
+
+    if(!layerDisplayName.empty()) {
+        layerDisplayName.pop_back();
```

Comment:
Why the pop_back()?

Reply:
the pop_back is just to remove the last "," when there are multiple layers

`[bob],[foo], `  

I pop back the last "," so it looks

`[bob],[foo]`

Reply:
Right, you've explained this to me once before...  A shortened version of your explanation would make a great code comment.